src/classes/UTIL_CascadeDeleteLookups_TDTM.cls

```diff
@@ -0,0 +1,131 @@
+ * their parent in the recycle bin.
+ * To enable cascade delete for a new relationship, add it to the "cascades" map. When adding a new object type, also
+ * add a corresponding trigger handler.
+ */
+public with sharing class UTIL_CascadeDeleteLookups_TDTM extends TDTM_Runnable {
+
+    /** @description Static list of sobjects for stashing objects for cascade deletion between before and after triggers */
+    private static List<SObject> forDelete = new List<SObject>();
+
+    /*******************************************************************************************************
+    * @description This map contains a keyset of object types that trigger cascade deletes, and values that store child
+    * objects that should be cascade deleted, and their lookup relationship to the parent object. Values are stored as a 
+    * list of strings; each string is of the format ObjectName.LookupFieldName, and multiple strings may be added to the 
+    * list
+    */
+    private final Map<SObjectType, List<String>> cascades = new Map<SObjectType, List<String>>{
+        Contact.SObjectType                      =>  new List<String>{'npe03__Recurring_Donation__c.npe03__Contact__c'},
+        Account.SObjectType                      =>  new List<String>{'npe03__Recurring_Donation__c.npe03__Organization__c'},
+        Opportunity.SObjectType                  =>  new List<String>{'Allocation__c.Opportunity__c'},
+        Campaign.SObjectType                     =>  new List<String>{'Allocation__c.Campaign__c'},
+        npe03__Recurring_Donation__c.SObjectType =>  new List<String>{'Allocation__c.Recurring_Donation__c'}
+    };
+
+    /*******************************************************************************************************
+    * @description Run method checks if there is a cascade deletion for the current trigger and runs the
+    * cascade method for each entry.
+    * @param listNew The list of sobjects from trigger new.
+    * @param listOld The list of sobjects from trigger old.
+    * @param triggerAction Which action triggered the current context. (Before or After; Insert, Update, 
+    * Delete, or Undelete.)
+    * @param objResult The field describe object for the current trigger set.
+    * @return dmlWrapper Objects that need to be inserted or updated after the processing of the trigger.
+    ********************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
+        TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+
+        if (cascades.containsKey(objResult.getSObjectType())) {
+            for (String childLookup : cascades.get(objResult.getSObjectType())) {
+                try {
+                    cascadeDelete(newlist, oldlist, childLookup, triggerAction);
+                } catch (Exception ex) {
+                    ERR_Handler.processError(ex, 'TDTM');
```

Comment:
I didn't think we'd want to throw an exception for an operation that would have succeeded but left behind orphaned records previously. Let's discuss.